functional_tests/android_listener_tests/test_listener_rest.py

```diff
@@ -0,0 +1,168 @@
+    with concurrent.futures.ThreadPoolExecutor(max_workers=len(device_defs)) as executor:
+
+        future_to_device_name = {
+            executor.submit(
+                create_listener,
+                target_device=device_def["target_device"],
+                local_port=device_def["local_port"],
+                apk_path=device_def["apk_path"],
+                activity=device_def["activity"],
+                reinstall=should_reinstall,
+            ): device_def["target_device"]
+            for device_def in device_defs
+        }
+
+        for future in concurrent.futures.as_completed(future_to_device_name):
+
+            name = future_to_device_name[future]
+            listener = future.result()
+
+            listeners[name] = listener
+            log.info("Listener created: {} {}".format(name, listener))
+
+    time.sleep(2)
+
+    emu_1 = listeners["emulator-5554"]
+    emu_2 = listeners["emulator-5556"]
+    emu_3 = listeners["emulator-5558"]
+    emu_4 = listeners["emulator-5560"]
+    emu_5 = listeners["emulator-5562"]
+
+    emu_1.verify_launched()
+    emu_2.verify_launched()
+    emu_3.verify_launched()
+    emu_4.verify_launched()
+    emu_5.verify_launched()
+
+    # Add docs on master device
+    emu_1_pusher = User(target=emu_1, db=db_name, name="emu1_doc_pusher", password="password", channels=["ABC"])
+    emu_1_pusher.add_docs(600, bulk=True)
+    emu_1_pusher.add_design_doc(
+        doc_id="view1",
+        content={
+            "views": {
+                "dt1": {
+                    "map": "function(doc) {if (doc.type == \'dt1\') {emit(doc._id, doc);}}"
+                },
+                "filters": {
+                    "dt1": "function(doc) {if (doc.type == \'dt1\') {return true} return false}"
+                }
+            }
+        }
+    )
+
+    # Start all replication
+    emu_1.start_pull_replication(emu_2.url, db=db_name)
+    emu_1.start_pull_replication(emu_3.url, db=db_name)
+    emu_1.start_pull_replication(emu_4.url, db=db_name)
+    emu_1.start_pull_replication(emu_5.url, db=db_name)
+
+    emu_1.start_push_replication(emu_2.url, db=db_name)
+    emu_1.start_push_replication(emu_3.url, db=db_name)
+    emu_1.start_push_replication(emu_4.url, db=db_name)
+    emu_1.start_push_replication(emu_5.url, db=db_name)
+
+    # Wait for all docs to push to emulators
+    time.sleep(30)
+
+    # TODO Assert each endpoint has 200 docs
+
+    # Stop all replication
+    emu_1.stop_pull_replication(emu_2.url, db=db_name)
+    emu_1.stop_pull_replication(emu_3.url, db=db_name)
+    emu_1.stop_pull_replication(emu_4.url, db=db_name)
+    emu_1.stop_pull_replication(emu_5.url, db=db_name)
+
+    emu_1.stop_push_replication(emu_2.url, db=db_name)
+    emu_1.stop_push_replication(emu_3.url, db=db_name)
+    emu_1.stop_push_replication(emu_4.url, db=db_name)
+    emu_1.stop_push_replication(emu_5.url, db=db_name)
+
+    # Wait for all replications to stop
+    time.sleep(30)
```

Comment:
@ashvindersingh This probably isn't the most elegant way to do this. I would think of this test as sort of a placeholder and was meant to provide a simulation of a scenario